addons/sale/models/sale_order.py

```diff
@@ -1287,14 +1287,15 @@ def _force_lines_to_invoice_policy_order(self):
 
     def payment_action_capture(self):
         """ Capture all transactions linked to this sale order. """
+        self.ensure_one()
         payment_utils.check_rights_on_recordset(self)
-        # In sudo mode because we need to be able to read on provider fields.
-        self.authorized_transaction_ids.sudo().action_capture()
+        # In sudo mode because we need to bypass the checks on the rights in the transactions.
```

Comment:
Useless diff, I could take that off as this isn't related to this PR. Rewrote that when I investigated the fact that it's to bypass the check made on `action_capture`/`action_void`, more than to read on the provider field (which indeed need the sudo, but that's not the question here, since the sudo is written and commented in the `action_capture`/`action_void` flow)